packages/davinci-client/src/lib/client.store.utils.ts

```diff
@@ -0,0 +1,30 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+import { configSlice } from './config.slice.js';
+import { nodeSlice } from './node.slice.js';
+import { davinciApi } from './davinci.api.js';
+import { ErrorNode, NextNode, StartNode, SuccessNode } from '../types.js';
+
+export function createClientStore() {
+  return configureStore({
+    reducer: {
+      config: configSlice.reducer,
+      node: nodeSlice.reducer,
+      [davinciApi.reducerPath]: davinciApi.reducer,
+    },
+    middleware: (getDefaultMiddleware) => {
+      return getDefaultMiddleware().concat(davinciApi.middleware);
+    },
+  });
+}
+
+type ClientStore = typeof createClientStore;
+
+export type RootState = ReturnType<ReturnType<ClientStore>['getState']>;
+
+export interface RootStateWithNode<T extends ErrorNode | NextNode | StartNode | SuccessNode>
```

Comment:
ooo, i'm curious about this.

Is there anything about redux that would cause a problem defining rootstate as a generic? 

This is probably whats giving us all the type inference.

I'm also curious if using `infer` here would help us get any extra type intelligence when a node may not be fully known.

I dont know off the top of my head, but maybe just leaving this here as a reminder in the future if we come across something


```ts
export interface RootStateWithNode<T extends ErrorNode | NextNode | StartNode | SuccessNode>
  extends RootState {
  node: T extends infer R ? R : never;
}
```